pandas/core/groupby/generic.py

```diff
@@ -1731,7 +1731,11 @@ def _wrap_aggregated_output(
         DataFrame
         """
         indexed_output = {key.position: val for key, val in output.items()}
-        columns = Index(key.label for key in output)
+        if self.axis == 0:
+            name = self._obj_with_exclusions.columns.name
+        else:
+            name = self._obj_with_exclusions.index.name
```

Comment:
is this code useful anywhere else? e.g. this would be fine as a property of the class L1734-37

Reply:
Did some regex searches, not currently that I can tell. Seems suitable for a property though, so might still make sense to add? Also note that with @WillAyd's help I was able to get it down to a single line, no branching. Let me know if it should be added.